lib/system/jssys.nim

```diff
@@ -495,11 +495,13 @@ proc negInt64(a: int64): int64 {.compilerproc.} =
 
 proc nimFloatToString(a: float): cstring {.compilerproc.} =
   ## ensures the result doesn't print like an integer, i.e. return 2.0, not 2
+  # print `-0.0` properly
   asm """
     function nimOnlyDigitsOrMinus(n) {
       return n.toString().match(/^-?\d+$/);
     }
-    if (Number.isSafeInteger(`a`)) `result` =  `a`+".0"
+    if (Number.isSafeInteger(`a`))
+      `result` = `a` === 0 && 1 / `a` < 0 ? "-0.0" : `a`+".0"
```

Comment:
@xflywind I just found out about `Object.is`, we should use it instead, it seems super useful here as well as in your `copySign` PR

```js
if(Object.is(`a`, -0)) `result` = "-0.0"
else if(Number.isSafeInteger(`a`)) `result` =  `a`+".0"
else { ...
```


ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is

we can add the polyfill provided there for IE, if needed

(it may have other uses too, see examples: `Object.is([], []);  // false`) which are hard to do without `Object.is`

Reply:
Is `Object.is` better than `a` === 0 && 1 / `a` < 0?

> we can add the polyfill provided there for IE, if needed
It emits more codes if we need to support IE.

I will think of it and look at whether scala.js uses `Object.is`.

Reply:
Should be more efficient